### ISSUE TITLE
Add ability to make popup trigger keyboard/tab focusable

### ISSUE THREAD

**Issue by kelleg1** [PopupMaker/Popup-Maker#474]
I originally posted this issue in the WordPress plugin support forum at https://wordpress.org/support/topic/keyboard-focus-on-links, where yogaman5020 seems to have noted issue #469 which seems similar to this, though I'm not sure if that has to do with popups or popup triggers. My issue is regarding popup triggers (the "links" that when clicked cause the popup to show).

Our organization is currently using version 1.6.6 of the plugin. The resolution to that issue and suggestion from yogaman5020 was to try v1.6.7-beta.2, which I did. I could not get popup triggers to be keyboard-focusable with either version of the plugin. I tried adding the tabindex="-1" as well as tabindex="1" by editing the HTML using Google Chrome's Inspector window to no avail.

By default, this plugin uses span elements for popup triggers, and I think the issue could be that span elements simply aren't focusable even with tabindex.

With version 1.6.7-beta.2, it is possible to change an individual popup trigger to use the anchor HTML tag instead of span. It is also possible to add a custom CSS class. Unfortunately however, without an href attribute on the anchor tag, it still isn’t focusable. Adding href="#" does make them focusable.

I suppose we could use JavaScript or PHP on our site to add href=”#” to the a tags, but that would probably involve editing our site’s theme, which we are trying to avoid. It would be nice if the plugin could do this when outputting the popup triggers' HTML.

Another issue is that the settings seem to need to be set for each individual popup trigger. This is a problem for our organization, since we have many editors in our WordPress who want to add popups to their pages, and they are probably not going to know/remember to change the options for each popup trigger that they add.

I propose the following:
- Add a setting to add href="#" to popup triggers, which will make them focusable, or make this the default behavior
- Add settings on the plugin's global settings page to allow choosing site-wide defaults for popup trigger options, including the HTML Tag and Custom CSS Class options
- Allow site-wide defaults to be overridden for an individual popup trigger in the existing popup trigger options

**Comment by danieliser** [comment]
@kelleg1 v1.7 changes the default from span to button.

I can get the href="#" added easily, my only concern is it will likely have adverse effects with any theme that adds smooth scrolling. (Salient, Avada) In that situation they return false on all links beginning with "#" which prevents our triggers from working.

What that means is if they had created a shortcode previously with the tag="a" parameter, when we add the href="#" part their triggers may stop working and require manually tracking down each to correct it in some way.

We could make it a new attribute. I have long considered changing the "tag" field to a select box with predefined options, if so then we could add additional fields. IE Choose Anchor tag, see link field, title attribute field etc.

May be too late to get that into v1.7.0 as its already going into public beta. But could sneak it into the first point update (1.7.1).

As for global settings, we likely won't be adding that. The reason why is simple, if we add them for shortcodes & triggers, then we will end up  adding them for cookies, conditions, display settings, extension meta etc. if that's the case why not just use any post duplicating tool to clone a default popup?

That said you can already do what you want, but the filter will change for v1.7 as we updated a lot of code to automate 90% of the form rendering/interactivity based on our array of form settings.

Basically you can filter the std key for any field in any of our forms (except popup theme editor until v1.8).

So in this case you want to modify this value: https://github.com/PopupMaker/Popup-Maker/blob/release/1.7/classes/Shortcode/PopupTrigger.php#L98

You do so via this filter: https://github.com/PopupMaker/Popup-Maker/blob/release/1.7/classes/Shortcode.php#L417

```
add_filter( 'pum_shortcode_fields', function ( $fields, PUM_Shortcode $shortcode ) {
   if ( $shortcode->tag() == 'popup_trigger' ) {
       $fields['options']['tag']['std'] = 'a';
   }

    return $fields;
}, 10, 2 );
```

Hope that helps.
